File: GestionListaTarea.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

// Función para mostrar el menú y obtener la opción del usuario
function mostrarMenu() {
  console.log("\nMenú:");
  console.log("1. Agregar tarea");
  console.log("2. Marcar tarea como completada");
  console.log("3. Mostrar tareas pendientes");
  console.log("4. Salir");
  rl.question("\nElige una opción:", (opcion) => {
    realizarAccion(opcion.trim());
  });
}

// Array para almacenar las tareas
let tareas = [];

// Realizar la acción según la opción elegida
function realizarAccion(opcion) {
  if (opcion === "1") {
    rl.question("Ingresa la nueva tarea:", (nuevaTarea) => {
      tareas.push({ tarea: nuevaTarea, completada: false });
      console.log("Tarea agregada correctamente.");
      mostrarMenu();
    });
  } else if (opcion === "2") {
    if (tareas.length === 0) {
      console.log("No hay tareas para marcar como completadas.");
      mostrarMenu();
    } else {
      console.log("Tareas Pendientes:");
      tareas.forEach((tarea, index) => {
        console.log(
          `${index + 1}. ${tarea.tarea} ${
            tarea.completada ? "(Completada)" : ""
          }`
        );
      });
      rl.question("Ingresa el número de tarea completada:", (numTarea) => {
        numTarea = parseInt(numTarea);
        if (numTarea >= 1 && numTarea <= tareas.length) {
          tareas[numTarea - 1].completada = true;
          console.log("Tarea marcada como completada.");
        } else {
          console.log("Número de tarea inválido.");
        }
        mostrarMenu();
      });
    }
  } else if (opcion === "3") {
    if (tareas.length === 0) {
      console.log("No hay tareas pendientes.");
    } else {
      console.log("\nTareas Pendientes:");
      tareas.forEach((tarea, index) => {
        if (!tarea.completada) {
          console.log(`${index + 1}. ${tarea.tarea}`);
        }
      });
    }
    mostrarMenu();
  } else if (opcion === "4") {
    console.log("¡Hasta luego!");
    rl.close();
  } else {
    console.log("Opción inválida. Inténtalo de nuevo.");
    mostrarMenu();
  }
}

// Iniciar el programa
mostrarMenu();

File: GestionListaTarea.py
# Función para mostrar el menú y obtener la opción del usuario
def mostrar_menu():
    print("\nMenú:")
    print("1. Agregar tarea")
    print("2. Marcar tarea como completada")
    print("3. Mostrar tareas pendientes")
    print("4. Salir")
    opcion = input("\nElige una opción: ")
    return opcion

# Lista para almacenar las tareas
tareas = []

# Bucle principal del programa
while True:
    opcion = mostrar_menu()

    if opcion == "1":
        nueva_tarea = input("Ingresa la nueva tarea: ")
        tareas.append({"tarea": nueva_tarea, "completada": False})
        print("Tarea agregada correctamente.")
    elif opcion == "2":
        if len(tareas) == 0:
            print("No hay tareas para marcar como completadas.")
        else:
            print("Tareas Pendientes:")
            for idx, tarea in enumerate(tareas):
                print(f"{idx + 1}. {tarea['tarea']} {'(Completada)' if tarea['completada'] else ''}")
            num_tarea = int(input("Ingresa el número de tarea completada: "))
            if 1 <= num_tarea <= len(tareas):
                tareas[num_tarea - 1]['completada'] = True
                print("Tarea marcada como completada.")
            else:
                print("Número de tarea inválido.")
    elif opcion == "3":
        if len(tareas) == 0:
            print("No hay tareas pendientes.")
        else:
            print("\nTareas Pendientes:")
            for idx, tarea in enumerate(tareas):
                if not tarea['completada']:
                    print(f"{idx + 1}. {tarea['tarea']}")
    elif opcion == "4":
        print("¡Hasta luego!")
        break
    else:
        print("Opción inválida. Inténtalo de nuevo.")
